core/src/main/scala/org/apache/spark/deploy/ApplicationDescription.scala

```diff
@@ -41,5 +41,8 @@ private[spark] class ApplicationDescription(
     new ApplicationDescription(
       name, maxCores, memoryPerSlave, command, appUiUrl, eventLogDir, eventLogCodec)
 
+  // only valid when spark.executor.multiPerWorker is set to true
+  var maxCorePerExecutor: Option[Int] = None
```

Comment:
This shouldn't be a `var` because we only ever set it in one place (when we create it). This should just go into the constructor.